### PR TITLE
Better failure reports for transition tests

### PR DESCRIPTION
**Pull request by martin-volf**
The filtering code keeps track of the current and new states, creating a list of transitions as they are performed; in case of one of known failure events it records the transition as failed. The list of transitions is then stored to operational CDB.

### CODE REVIEW

File: python/drned_xmnr/op/filtering/events.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    def produce_line(self):
+        return self.line
+
+
+class InitialPrepareEvent(LineOutputEvent):
+    def __init__(self):
+        super(InitialPrepareEvent, self).__init__('Prepare the device')
+
+
+class InitStates(LineOutputEvent):
```
Comment (jojohans) [inline python/drned_xmnr/op/filtering/events.py:39]: I've been looking at this for several weeks. Is there a reason the some event names end in "Event" and others dosen't?

**Review by jojohans — APPROVED** [review]
I was wondering if it would make sense to treat "can't restore config using NETCONF you have to use the CLI" errors separately.

On the other hand we will hope fully fix it this week or the week after, maybe not worth it.

File: python/drned_xmnr/op/filtering/events.py (hunk 1 above)
Comment (martin-volf) [reply python/drned_xmnr/op/filtering/events.py:39]: I guess I started naming them all "...Event" and eventually became lazy, or something like that.

**Comment by jojohans** [comment]
Martin,

Since you haven't merged this yet. I got an "Silence timeout, terminating process" error in the log file... resulting in "failure conversion failed; is device driver" in the CLI. I don't think it's what we want.

File: python/drned_xmnr/op/filtering/events.py (hunk 1 above)
Comment (martin-volf) [reply python/drned_xmnr/op/filtering/events.py]: fixed

File: python/drned_xmnr/op/filtering/cort.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@coroutine
+def filter_sink(writer):
+    while True:
+        item = yield
+        if isinstance(item, str):
+            # the writer needs full line
+            writer(item + '\n')
+
+
+class Closeable(object):
```
[the remark this answers is not in the document]
Comment (martin-volf) [reply python/drned_xmnr/op/filtering/cort.py]: No, that's not needed. Removed.

**Comment by martin-volf** [comment]
I see that... the message is incomplete, I obviously had something in mind, but cannot recall what. Let's deal with this separately, there's more to improve in failure reports for the import/conversion actions.

**Pull request merged**